scripts/crawler.js

```diff
@@ -18,8 +18,27 @@ const V8_NATIVE_CALL_REPLACEMENT_PREFIX = '__v8_native_remainder'
 const V8_NATIVE_CALL_REPLACEMENT_REGEX = /__v8_native_remainder(\w+\(\S*?|\s*\))/gm
 const V8_NATIVE_CALL_FLAGS_COMMENT_REGEX = /\/\/\s*Flags:.*(--allow-natives-syntax)+/gm
 
-const GLOBAL_METHODS =
-  ";(function(globals){globals._ddiast = globals._ddiast || {plusOperator(res) {return res;}};}((1,eval)('this')));"
+const CSI_METHODS = [
+  { src: 'plusOperator', operator: true },
+  { src: 'substring' },
+  { src: 'trim' },
+  { src: 'trimStart' },
+  { src: 'trimEnd' },
+  { src: 'trimLeft' },
+  { src: 'trimRight' },
+  { src: 'toLowerCase' },
+  { src: 'toLocaleLowerCase' },
+  { src: 'toUpperCase' },
+  { src: 'toLocaleUpperCase' },
+  { src: 'replace' },
+  { src: 'replaceAll' },
+  { src: 'slice' },
+  { src: 'concat' }
+]
+
+const GLOBAL_METHODS_TEMPLATE = `;(function(globals){
+  globals._ddiast = globals._ddiast || { __CSI_METHODS__ };
+}((1,eval)('this')));`
```

Comment:
why `(1,eval)('this')` instead of `this` or `eval('this')`?

Reply:
"Well, the Ecma spec considers a reference to eval to be a "direct eval call", but an expression that merely yields eval to be an indirect one -- and indirect eval calls are guaranteed to execute in global scope."

I tried different options and some test only worked if the injection was made with `(1,eval)('this')`.